golem-worker-executor-base/src/storage/cassandra.rs

```diff
@@ -0,0 +1,675 @@
+// Copyright 2024 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use bytes::Bytes;
+use futures::StreamExt;
+
+use golem_common::config::CassandraConfig;
+use golem_common::metrics::db::{record_db_failure, record_db_success};
+use scylla::batch::{Batch, BatchType};
+use scylla::prepared_statement::PreparedStatement;
+use scylla::query::Query;
+use scylla::serialize::row::SerializeRow;
+use scylla::transport::errors::QueryError;
+use scylla::FromRow;
+use scylla::{transport::session::PoolSize, Session, SessionBuilder};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::iter::repeat;
+use std::time::Instant;
+use std::{num::NonZeroUsize, sync::Arc};
+
+#[derive(Debug, Clone)]
+pub struct CassandraSession {
+    pub session: Arc<Session>,
+    keyspace: String,
+    set_tracing: bool,
+}
+
+impl CassandraSession {
```

Comment:
this struct and impl is still specific to our key value store schema, and not a generic cassandra session. anything that is using the kv_store table etc. tables should be named and placed as part of the KVStore implementation, and only things that are using generic cassandra primitives should be named as such cassandra

Reply:
Yes, as I tried to explain yesterday, this is a common struct as CassandraLabelledApi, which will contain all the kv and indexed storage queries, tables, and functions.
This PR is KV related, and I have the Indexed related changes for this file and other files. 
Once this PR is approved I'll create the Indexed storage PR, probably you will see what I mean at that time.

Reply:
What we are trying to ask is to follow the structure of how it was done for Redis:

- One layer wraps the 3rd party library and adds metrics and logging - without adding any higher level operations (no actual queries, just the same operations provided by the 3rd party library, wrapped)
- KV store implementation built on this, only containing KV store related queries
- Indexed store implementations built on this, only containing indexed store related queries

Reply:
I see, probably we can create another ticket to fix sqlite implementation as it is done in the same way as Cassandra.
There is a struct like this
`SqliteLabelledApi {
    svc_name: &'static str,
    api_name: &'static str,
    pool: SqlitePoolx,
}`
to avoid passing common parameters like `svc_name` & `api_name` for every method to the metrics and logging functionality, so this Labelled is holding the 3 storage kind of queries for simplicity, but it seems we are going to change that in Cassandra, then I think we need to change also in sqlite.

WDYT ?

Reply:
yes, the sqlite one should also use similar naming and structuring